src/main/scala/viper/server/vsi/VerificationServer.scala

```diff
@@ -170,16 +169,15 @@ trait VerificationServer extends Unpacker {
         } else {
           sender() ! s"Job #$id has already been finalized."
         }
-      case VerificationProtocol.Verify(task, queue, publisher) =>
+      case VerificationProtocol.Verify(task, queue, publisher, executor) =>
         resetVerificationTask()
-        sender() ! startJob(task, queue, publisher)
+        sender() ! startJob(task, queue, publisher)(executor)
       case msg =>
         throw new Exception("Main Actor: unexpected message received: " + msg)
     }
 
-    private def startJob(task: Thread, queue: SourceQueueWithComplete[Envelope], publisher: Publisher[Envelope]): JobHandle = {
-      _verificationTask = task
-      _verificationTask.start()
+    private def startJob(task: VerificationTask, queue: SourceQueueWithComplete[Envelope], publisher: Publisher[Envelope])(executor: VerificationExecutionContext): JobHandle = {
```

Comment:
One could probably make ```executor``` an implicit parameter.

Reply:
One could but this in my understanding this would only affect the functions body in which I do not need an implicit executor (the call to `submit` requires an explicit receiver since Scala does not seem to accept an implicit receiver)